src/lib/npfg/npfg.cpp

```diff
@@ -101,7 +101,10 @@ void NPFG::guideToPath(const Vector2f &ground_vel, const Vector2f &wind_vel, con
 	// speed violations and/or high wind conditions in general
 	air_vel_ref_ = refAirVelocity(wind_vel, bearing_vec_, wind_cross_bearing,
 				      wind_dot_bearing, wind_speed, min_ground_speed_ref_);
-	airspeed_ref_ = air_vel_ref_.norm();
+
+	if (PX4_ISFINITE(air_vel_ref_.norm())) {
+		airspeed_ref_ = air_vel_ref_.norm();
+	}
```

Comment:
I guess this means that it will just use the previous airspeed reference?

Reply:
Yes correct, in that case it would just use the previous `airspeed_ref_`. 
@tstastny actually, in this case it would be good if `airspeed_ref_ = airspeed_nom_`, but this seems to only happen in the currently unused function `navigateLevelFlight()`. Curious for your inputs on how we should make it robust.

Reply:
If we sanity check position setpoints on the module interface (see other comment) - I don't think we'll need all these embedded `ISFINITE` checks. For NPFG (library) itself, if we really wanted to be safe, the interfaces could also be checked (e.g. the waypoints or loiter points etc on interface to the navigation methods), but then every library method should be checking for NaNs on every single input if that is the case. Maybe module interfaces are the better minimal place to sanity check for the moment without needing to adapt every single library on the system?

Then ideally we would have unit tests on the methods that manipulate the already sanitized data creating setpoints (e.g. airspeed) that go to the libraries or other modules.

If the position setpoint is invalid - i dont think NPFG position oriented methods should even be called from the position control module. we should define some other behavior instead of leaving it to chance what the last setpoint was etc

Reply:
After discussing with @tstastny I removed the sanitization from NPFG, and instead added some for TECS: d28954106940ec31c89cbd1745db96c82ad32d25